zmq.hpp

```diff
@@ -399,16 +399,36 @@ class message_t
             throw error_t();
     }
 
+    // overload set of string-like types and generic containers
 #if defined(ZMQ_CPP11) && !defined(ZMQ_CPP11_PARTIAL)
+    // NOTE this constructor will include the null terminator
+    // when called with a string literal.
+    // An overload taking const char* can not be added because
+    // it would be preferred over this function and break compatiblity.
     template<class Range,
              typename = typename std::enable_if<
                detail::is_range<Range>::value
                && ZMQ_IS_TRIVIALLY_COPYABLE(detail::range_value_t<Range>)
                && !std::is_same<Range, message_t>::value>::type>
+    ZMQ_DEPRECATED(
+        "from 4.7.0, use constructors taking iterators, (pointer, size) or strings instead")
```

Comment:
I think in the future with C++20 ranges, it will seem odd to recommend use of an overload accepting a pair of iterators rather than a range. Could we add an overload `template<size_t N, > explicit message_t(const char (&rng)[N])` and deprecate just that?

Reply:
Yes good point.

Reply:
Thanks!